resource-managers/yarn/src/main/resources/META-INF/services/org.apache.spark.deploy.security.ServiceCredentialProvider

```diff
@@ -0,0 +1,3 @@
+org.apache.spark.deploy.yarn.security.YARNHadoopFSCredentialProvider
```

Comment:
This worried me a little bit. Because I'm pretty sure `ServiceLoader` will look at *both* files (this one and the one in core), and when running in YARN you'll end up with way too many providers.

So you should verify that behavior, and the best way to do it is to write a unit test.